### PR TITLE
Use long when setting REG_QWORD

### PR DESCRIPTION
**Pull request by twangboy**
### What does this PR do?
Uses long for REG_QWORD values in Py2

### What issues does this PR fix or reference?
https://github.com/saltstack/salt/issues/50039

### Tests written?
Yes

### Commits signed with GPG?
Yes

### CODE REVIEW

**Comment by twangboy** [comment]
Regarding the lint failure... It is suggesting that I use `int` instead of `long` in Py2. It has to be `long` on Py2 for the Windows API call that occurs later on.

**Comment by rallytime** [comment]
@twangboy Can you put a pylint disable for that check then? We can't merge this if lint is red. :)

**Comment by lorengordon** [comment]
Conversion guide seems to suggest `from builtins import int`... Try that yet?

http://python-future.org/compatible_idioms.html#long-integers

**Comment by dwoz** [comment]
@twangboy Lets see if @lorengordon 's comment works before merging this.

**Review by dwoz — CHANGES_REQUESTED** [review]
@twangboy Lets see if @lorengordon's comment works before merging this.

**Comment by twangboy** [comment]
@lorengordon Your suggestion requires adding another dependency to Salt. It doesn't work on Py2 by default. You have to `pip install future`. So, rather than add another dependency, I think I would rather just use this fix.

**Comment by lorengordon** [comment]
@twangboy Ahh, that's true. I already use `future` everywhere I need 2/3 compatibility because it is so darn handy for these things (much like `six`). Figured you were also. Oh well.

File: salt/utils/win_reg.py
Comment (rallytime) [inline salt/utils/win_reg.py]: What about using `six.integer_types` here?

Seems nicer that disabling the lint error.

https://pythonhosted.org/six/#six.integer_types

Comment (twangboy) [reply salt/utils/win_reg.py]: I tried that, but it seems to only be useful for detecting types not casting.
```
>>> from salt.ext import six
>>> long(7)
7L
>>> six.integer_types(7)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: 'tuple' object is not callable
>>> isinstance(7, six.integer_types)
True
>>> isinstance(7L, six.integer_types)
True
>>>
```

Comment (twangboy) [reply salt/utils/win_reg.py]: @rallytime You want I should add `future` as a dependency instead? That works and is quite clean.

Comment (rallytime) [reply salt/utils/win_reg.py]: Ah, right, OK. I think this is fine as-is. I'd rather not add the dependency. Your current check works.

**Review by rallytime — APPROVED** [review]

**Comment by cachedout** [comment]
@dwoz Do you want to come back and look at this again?

**Review by dwoz — APPROVED** [review]

**Comment by cachedout** [comment]
We didn't get Windows test results on that run so I've restarted the test suite here.

**Comment by twangboy** [comment]
Rebased

**Pull request merged**
